src/common/utils/time/time.ts

```diff
@@ -59,44 +49,44 @@ export function exclusiveToInclusiveEnd(exclusiveEnd: Date): Date {
   return new Date(exclusiveEnd.valueOf() - 1);
 }
 
-export function formatTimeRange(timeRange: TimeRange, timezone: Timezone, displayYear: DisplayYear): string {
-  const { start, end } = timeRange;
-  const startWallTime = moment.tz(start, timezone.toString());
-  const endWallTime = moment.tz(end, timezone.toString());
-  const endWallTimeInclusive = getEndWallTimeInclusive(end, timezone);
-
-  let showingYear = true;
-  let formatted: string;
-  if (startWallTime.year() !== endWallTimeInclusive.year()) {
-    formatted = [startWallTime.format(FORMAT_WITH_YEAR), endWallTimeInclusive.format(FORMAT_WITH_YEAR)].join(" - ");
-  } else {
-    showingYear = displayYear === DisplayYear.ALWAYS ||
-      (displayYear === DisplayYear.IF_DIFF && !isCurrentYear(endWallTimeInclusive.year(), timezone));
-    const fmt = showingYear ? FORMAT_WITH_YEAR : FORMAT_WITHOUT_YEAR;
-    if (startWallTime.month() !== endWallTimeInclusive.month() || startWallTime.date() !== endWallTimeInclusive.date()) {
-      formatted = [startWallTime.format(FORMAT_WITHOUT_YEAR), endWallTimeInclusive.format(fmt)].join(" - ");
-    } else {
-      formatted = startWallTime.format(fmt);
-    }
+function formatDate(start: Moment, end: Moment, showingYear: boolean): string {
+  if (start.year() !== end.year()) {
+    return [start.format(FORMAT_WITH_YEAR), end.format(FORMAT_WITH_YEAR)].join(" - ");
   }
+  const fmt = showingYear ? FORMAT_WITH_YEAR : FORMAT_WITHOUT_YEAR;
+  if (start.month() !== end.month() || start.date() !== end.date()) {
+    return [start.format(FORMAT_WITHOUT_YEAR), end.format(fmt)].join(" - ");
+  }
+  return start.format(fmt);
+}
 
-  if (startWallTime.hour() || startWallTime.minute() || endWallTime.hour() || endWallTime.minute()) {
-    formatted += (showingYear ? " " : ", ");
-
-    let startTimeStr = formatTimeOfDay(startWallTime).toLowerCase();
-    const endTimeStr = formatTimeOfDay(endWallTime).toLowerCase();
+function formatTime(start: Moment, end: Moment): string {
+  if (!start.hour() && !end.hour() && !start.minute() && !end.minute()) return null;
+  return `${start.format(FORMAT_TIME_OF_DAY_WITH_MINUTES)} - ${end.format(FORMAT_TIME_OF_DAY_WITH_MINUTES)}`;
+}
 
-    if (startTimeStr === endTimeStr) {
-      formatted += startTimeStr;
-    } else {
-      if (startTimeStr.substr(-2) === endTimeStr.substr(-2)) {
-        startTimeStr = startTimeStr.substr(0, startTimeStr.length - 2);
-      }
-      formatted += [startTimeStr, endTimeStr].join("-");
-    }
+function showYear(displayYear: DisplayYear, startYear: number, endYear: number): boolean {
```

Comment:
I don't like this function, when I'm reading the signature I don't know what to expect as a result.

Reply:
It should decide if we want to display year in the date. Maybe something like "shouldDisplayYear(startYear, endYear, displayYear)"?